File: WebAllGrade/src/main/java/com/satanko/weballgrade/data/model/JsonLesson.java
/************************************************************
 *  Copyright (c) 2013 Philipp Sommersguter, Julian Tropper *
 *                                                          *
 *  See the file LICENCE.txt for copying permission.        *
 ***********************************************************/
package com.satanko.weballgrade.data.model;

import com.satanko.weballgrade.data.dao.impl.hibernate.HibernateLessonDAO;
import java.util.ArrayList;

/**
 * This class acts as a model class for holding data in the application.
 * It represents a variation of {@link Lesson} as posted from ExcelAllGrade in {@link HibernateLessonDAO#createOrUpdateLessons(java.util.ArrayList, java.util.ArrayList)}.
 * 
 * @author Julian Tropper (julian.tropper@gmail.com)
 * ATTENTION: If not otherwise specified, the author(s) of the class is/are also responsible for all methods.
 * 
 * @version 1.0
 */
public class JsonLesson {
    private String classToken;
    private String subjectToken;
    private ArrayList<String> teacherList = new ArrayList<>();
    
    public JsonLesson (String classToken, String subjectToken)
    {
        this.classToken = classToken;
        this.subjectToken = subjectToken;
    }
    
    public void addTeacher(String teacherToken)
    {
        teacherList.add(teacherToken);
    }

    public String getClassToken() {
        return classToken;
    }

    public void setClassToken(String classToken) {
        this.classToken = classToken;
    }

    public String getSubjectToken() {
        return subjectToken;
    }

    public void setSubjectToken(String subjectToken) {
        this.subjectToken = subjectToken;
    }

    public ArrayList<String> getTeacherList() {
        return teacherList;
    }

    public void setTeacherList(ArrayList<String> teacherList) {
        this.teacherList = teacherList;
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    public boolean equals(Object obj) {
        JsonLesson lesson = (JsonLesson) obj;
        
        String classToken = lesson.classToken;
        String subjectToken = lesson.subjectToken;
        ArrayList<String> teacherTokens = lesson.getTeacherList();
        
        if(this.classToken.equals(classToken) && this.subjectToken.equals(subjectToken))
        {
            for (String teacherToken : teacherTokens)
            {
                if (!this.teacherList.contains(teacherToken))
                {
                    return false;
                }
            }
        }
        else 
        {
            return false;
        }
        
        return true;
    }
}

File: ExcelAllGrade/ExcelAllGrade/model/JsonLesson.cs
//  ************************************************************
//  *  Copyright (c) 2013 Philipp Sommersguter, Julian Tropper *
//  *                                                          *
//  *  See the file LICENCE.txt for copying permission.        *
//  ************************************************************ 

using Newtonsoft.Json;
using System.Collections.Generic;

namespace ExcelAllGrade.model
{
    /// <summary>
    /// This class represents a <c>JsonLesson</c>.
    /// </summary>
    public class JsonLesson
    {
        /// <summary>
        /// Gets or sets the <c>SubjectToken</c> of the <see cref="JsonLesson"/>.
        /// </summary>
        public string SubjectToken { get; set; }
        /// <summary>
        /// Gets or sets the <c>ClassToken</c> of the <see cref="JsonLesson"/>.
        /// </summary>
        public string ClassToken { get; set; }
        /// <summary>
        /// Gets or sets the <c>Teachers</c> of the <see cref="JsonLesson"/>.
        /// </summary>
        public List<Teacher> Teachers { get; set; }

        /// <summary>
        /// Initializes a new instance of <see cref="JsonLesson"/>.
        /// </summary>
        /// <param name="LessonClassToken">The <see cref="ClassToken"/> of the <see cref="JsonLesson"/>.</param>
        /// <param name="LessonSubjectToken">The <see cref="SubjectToken"/> of the <see cref="JsonLesson"/>.</param>
        /// <param name="LessonTeacherList">The <see cref="Teachers"/> of the <see cref="JsonLesson"/>.</param>
        [JsonConstructor]
        public JsonLesson(string LessonClassToken, string LessonSubjectToken, List<Teacher> LessonTeacherList)
        {
            this.SubjectToken = LessonSubjectToken;
            this.ClassToken = LessonClassToken;
            this.Teachers = LessonTeacherList;
        }

        /// <summary>
        /// Initializes a new instance of <see cref="JsonLesson"/>. <see cref="Teachers"/> will be initialized as an empty <see cref="List{T}"/>.
        /// </summary>
        /// <param name="LessonClassToken">The <see cref="ClassToken"/> of the <see cref="JsonLesson"/></param>
        /// <param name="LessonSubjectToken">The <see cref="SubjectToken"/> of the <see cref="JsonLesson"/></param>
        public JsonLesson(string LessonClassToken, string LessonSubjectToken)
        {
            this.SubjectToken = LessonSubjectToken;
            this.ClassToken = LessonClassToken;
            this.Teachers = new List<Teacher>();
        }

        /// <summary>
        /// Adds a new <see cref="ExcelAllGrade.model.Teacher"/> to <see cref="Teachers"/>.
        /// </summary>
        /// <param name="teacher">The <see cref="ExcelAllGrade.model.Teacher"/> to add</param>
        public void addTeacher(Teacher teacher)
        {
            this.Teachers.Add(teacher);
        }
    }
}
